Adaptors/RabbitMQ/src/PushQueueStorage.cs

```diff
@@ -80,29 +83,32 @@ private void PushMessages(IEnumerable<MessageData> messages,
     var queueArgs = new Dictionary<string, object>
                     {
                       {
-                        "x-max-priority", Options!.MaxPriority
+                        "x-max-priority", Options.MaxPriority
                       },
                       {
                         "x-queue-mode", "lazy" // queue will try to move messages to disk as early as practically possible
                       },
                     };
 
-    ConnectionRabbit.Channel!.QueueDeclare(partitionId,
-                                           false, /* to survive broker restart */
-                                           false, /* used by multiple connections */
-                                           false, /* not deleted when last consumer unsubscribes (if it has had one) */
-                                           queueArgs);
+    (await ConnectionRabbit.GetConnectionAsync(cancellationToken)
+                           .ConfigureAwait(false)).QueueDeclare(partitionId,
+                                                                false, /* to survive broker restart */
+                                                                false, /* used by multiple connections */
+                                                                false, /* not deleted when last consumer unsubscribes (if it has had one) */
+                                                                queueArgs);
 
     foreach (var msg in messages)
     {
-      var basicProperties = ConnectionRabbit.Channel!.CreateBasicProperties();
+      var basicProperties = (await ConnectionRabbit.GetConnectionAsync(cancellationToken)
+                                                   .ConfigureAwait(false)).CreateBasicProperties();
       basicProperties.Priority = Convert.ToByte(priority);
       basicProperties.MessageId = Guid.NewGuid()
                                       .ToString();
-      ConnectionRabbit.Channel.BasicPublish("",
-                                            partitionId,
-                                            basicProperties,
-                                            Encoding.UTF8.GetBytes(msg.TaskId));
+      (await ConnectionRabbit.GetConnectionAsync(cancellationToken)
+                             .ConfigureAwait(false)).BasicPublish("",
+                                                                  partitionId,
+                                                                  basicProperties,
+                                                                  Encoding.UTF8.GetBytes(msg.TaskId));
```

Comment:
Is there a BasicPublishAsync? As you already have changed the method to async, it might be the right moment to have an async publish.

Reply:
No async available with this Package